source/operations/cert-manager.rst

```diff
@@ -0,0 +1,577 @@
+.. _minio-certmanager:
+
+============
+cert-manager
+============
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 1
+
+TLS certificate management with cert-manager
+--------------------------------------------
+
+`cert-manager <https://cert-manager.io/>`__ works as a controller for managing certificates within Kubernetes clusters.
+cert-manager obtains valid certificates from an ``Issuer`` or ``ClusterIssuer`` and can renew certificates prior to expiration.
+
+The MinIO Operator supports using cert-manager for certificates as an alternative to the MinIO Operator managing certificates for itself and its tenants.
+
+On a Kubernetes cluster, cert-manager requires a global level ``ClusterIssuer`` to generate intermediary ``Issuers`` and certificates.
+At the namespace level, cert-manager issues certificates derived from an ``Issuer``.
+A ``ClusterIssuer`` can issue certificates for multiple namespaces.
+An ``Issuer`` only issues certificates for its own namespace.
+
+To learn more about cert-manager Issuers, refer to the `Issuer documentation <https://cert-manager.io/docs/concepts/issuer/>`__.
+
+Below is a logical view of a Kubernetes cluster with four namespaces:
+
+- ``minio-operator``
+- ``tenant-1``
+- ``tenant-2``
+- ``other-namespace``
+
+.. image:: /images/k8s/cert-manager-namespaces.svg
+   :width: 600px
+   :alt: A graphic depiction of a kubernetes cluster with four namespaces. Each namespace in its own white box. One is labeled minio-operator with contents two sts content items. A second is labeled tenant-1 with a MinIO logo and four drives of a pool. A third is labeled tenant-2 with similar contents to the tenant-1 box. The fourth is labeled other namespace with contents of other pods.
+   :align: center
+
+This guide shows you how to set up a different Certificate Authority (CA) in each namespace.
+Each namespace's CA references the global ``Cluster Issuer``.
+
+
+The following graphic depicts how various namespaces make use of either an ``Issuer`` or ``ClusterIssuer`` type.
+
+- cert-manager is installed in the ``cert-manager`` namespace, which does not have either an ``issuer`` or a ``ClusterIssuer``.
+- The ``default`` namespace receives the global ``Cluster Issuer``.
+- Each tenant's namespace receives a local ``Issuer``.
+- The ``minio-operator`` namespace receives a local ``Issuer``. 
+  More about services that require TLS certificates in the ``minio-operator`` namespace are covered :ref:`below <minio-operator-services-with-cert-manager>`.
+
+.. image:: /images/k8s/cert-manager-cluster.svg
```

Comment:
Suggest modifying this to be a root-leaf type diagram where certmanager is the 'root' and then each 'branch' leads to the leaf namespaces.

Reply:
I guess the question is do we want to show how cert-manager works from the view point of the cluster, or how the certificates work within a cert-manager world.

Reply:
Something more like this?

![certmanagerissuers drawio](https://github.com/user-attachments/assets/a20ffbea-c480-47dc-a7d5-179e6c0bbe48)

Reply:
I think for the purposes of our docs, the focus should be on the ClusterIssuer > Issuer relationship. The original diagram was focused more on namespaces.

Reply:
Agreed - less is more, and I think focusing on how cert-manager works with MinIO + Tenants is the critical bit.